emsdk

```diff
@@ -1395,7 +1395,8 @@ class Tool:
         success = git_clone_checkout_and_pull(url, self.installation_path(), self.git_branch)
       elif url.endswith('zip') or url.endswith('.tar') or url.endswith('.gz') or url.endswith('.xz') or url.endswith('.tbz2'):
         download_even_if_exists = (self.id == 'vs-tool')
-        success = download_and_unzip(url, self.installation_path(), download_even_if_exists, filename_prefix=self.zipfile_prefix if hasattr(self, 'zipfile_prefix') else '')
+        filename_prefix = self.zipfile_prefix if hasattr(self, 'zipfile_prefix') else ''
```

Comment:
I think this can be more simple written as:
```
filename_prefix = getattr(self, 'zipfile_prefix', '')
```

Reply:
I did not know `getattr` had a default field! TIL

I was thinking earlier that I'd personally prefer
```python
filename_prefix = ''
if hasattr(self, 'zipfile_prefix'):
  filename_prefix = self.zipfile_prefix
```
but that's just me not liking python's `B if A else C` way of reading an `A ? B : C` so I didn't mention it (also because it was essentially moving existing code)
But getattr default is better still, so yes please

Reply:
Nice, thanks, updated.